README

```diff
@@ -26,3 +26,79 @@ systems.  This utility is used by util/genliso and util/gensdsk.  You
 will need to determine whether your use case requires these utilities.
 
 For any more detailed instructions, see http://ipxe.org
+
+## Updating
+
+Occasionally we'll want to sync with upstream, e.g., to pick up nic card
+driver updates. This documents that process.
+
+1. First you need an issue ID either in Jira or github for the sync up.
+2. Add a remote for the upstream ipxe repo if you don't already have one.
+
+        git remote add upstream https://github.com/ipxe/ipxe
+
+3. Add a remote tracking branch for the upstream repo and fetch their
+   changes.
+
+        git checkout -b upstream upstream/master
+        git fetch upstream
+
+4. Create a branch for the issue and merge from upstream.
+
+        git checkout -b <ISSUE_ID>
+        git merge upstream
+
+    Change the commit message to reflect Joyent's commit message format.
+
+5. Push the issue branch to Joyent.
+
+        git push -u origin HEAD:<branchname>
+
+    Jenkins will automatically create a build for you and push the build
+    artifacts to Manta and the experimental channel on updates.joyent.com.
+
+6. Once the jenkins build is done you'll need to update sdc-booter, so
+   cd to wherever you have a clone of that repo.
+
+        cd <wherever you keep your repos>/sdc-booter
+
+7. Again, create a branch for the issue.
+
+        git checkout -b <ISSUE_ID>
+
+8. Use the `update_ipxe` script. This will update `tftpboot/snponly.efi` and
+   `tfptboot/undionly.kpxe`. Again, commit the changes and push the branch
+   up.
+
+        ./tools/update_ipxe <ISSUE_ID>
+	git add tftpboot
+	git commit
+	git push -u origin HEAD:<branchname>
+
+    And again, Jenkins will automatically create a build and push it to
+    the experimental channel.
+
+You're done. After testing, get code review, integration approval, then merge.
```

Comment:
We should mention that you'll need help from an admin to allow a push of the multiple changesets

Reply:
Something like this here?

> Note: You need to have write permission to Joyent's repo(s) for this step. If you're an outside contributor push it to your own fork and open a pull request.

Reply:
No, even normal ones like us will need get the settings changed as the repo should be configured to only allow single-commit pushes

Reply:
Ok, I've discussed with @trentm and we'll enable both merge and rebase operations, and it'll be up to the person conducting the merge to make sure to do the right thing. I've added a comment to the readme about this.

In the future, we can work out a safer way to handle this, but this should suffice for now.